subsys/net/lib/shell/iface.c

```diff
@@ -274,6 +274,11 @@ static void iface_cb(struct net_if *iface, void *user_data)
 	   net_if_get_device(iface) ? net_if_get_device(iface)->name : "<?>",
 	   net_if_get_device(iface));
 
+	PR("Status    : oper=%s, admin=%s, carrier=%s\n",
+	   net_if_oper_state2str(net_if_oper_state(iface)),
+	   net_if_is_admin_up(iface) ? "UP" : "DOWN",
+	   net_if_is_carrier_ok(iface) ? "ON" : "OFF");
```

Comment:
Shall we also print dormant flag here (`net_if_is_dormant()`)? Could be useful for Wi-Fi interfaces. This will provide all information needed to tell if interface is operational or not.

Reply:
I had that printed but it is kind of redundant information as the oper status already prints the DORMANT if we are dormant.

Reply:
Well, one could argue the same for carrier state ("LOWERLAYERDOWN"), but ok, not strong push, maybe that isn't so obvious.